server/src/domain/system-config/system-config.service.ts

```diff
@@ -53,6 +73,48 @@ export class SystemConfigService {
     this.core.addValidator(validator);
   }
 
+  async handleImmichLatestVersionAvailable() {
+    if (this.disableCheckLatestVersion) {
+      return true;
+    }
+    const proxy =
+      this.proxyHost && this.proxyPort && this.proxyProtocol
+        ? {
+            protocol: this.proxyProtocol,
+            port: this.proxyPort,
+            host: this.proxyHost,
+          }
+        : false;
+
+    try {
+      const { data } = await axios.get<GithubRelease>(
+        'https://api.github.com/repos/immich-app/immich/releases/latest',
+        {
+          headers: {
+            Accept: 'application/vnd.github.v3+json',
+          },
+          proxy,
+        },
+      );
+
+      const [, major, minor, patch] = data.tag_name.match(/v(\d+)\.(\d+)\.(\d+)/) || [];
+
+      const temp = new ServerVersion(major, minor, patch);
+
+      if (temp.toString().localeCompare(serverVersion.toString(), undefined, { numeric: true }) === 1) {
+        this.logger.debug(`New version Immich version available : ${temp.toString()}`);
+        this.availableVersion = temp;
+      } else {
+        this.logger.debug('No new Immich version available detected');
+      }
```

Comment:
This code probably belongs in some utility file, which can have some separate tests for it directly.